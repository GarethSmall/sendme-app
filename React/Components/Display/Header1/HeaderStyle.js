// @flow
import { StyleSheet } from 'react-native';
import AppTheme from '../../../AppTheme/AppTheme';

export default (HeaderStyle = StyleSheet.create({
  header: {
    fontFamily: AppTheme.fontFamily,
    fontWeight: 'bold',
    fontSize: AppTheme.header1,
    color: AppTheme.whiteColor,
    padding: AppTheme.largeSize,
  },
}));
