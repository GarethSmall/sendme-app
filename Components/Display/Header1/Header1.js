// @flow
import React, { PureComponent } from 'react';
import { Text } from 'react-native';
import HeaderStyle from './HeaderStyle';

export default class Header1 extends PureComponent {
  render() {
    return (
      <Text style={HeaderStyle.header}>
        {this.props.children}
      </Text>
    );
  }
}