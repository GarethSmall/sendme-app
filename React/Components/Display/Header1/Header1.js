// @flow
import React, { PureComponent } from 'react';
import { Text } from 'react-native';
import HeaderStyle from './HeaderStyle';

type Props = {
  style? : Object,
};

export default class Header1 extends PureComponent<Props> {
  render() {
    const { style } = this.props;
    return <Text style={[HeaderStyle.header, style]}>{this.props.children}</Text>;
  }
}
