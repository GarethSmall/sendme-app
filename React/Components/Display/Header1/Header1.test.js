// @flow
import React from 'react';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
import shallow from 'enzyme/shallow';
import Header1 from './';
const { describe, test, expect } = global;

Enzyme.configure({ adapter: new Adapter() });

describe('Header1 ', () => {
  test('should render', () => {
    const component = shallow(<Header1 />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
