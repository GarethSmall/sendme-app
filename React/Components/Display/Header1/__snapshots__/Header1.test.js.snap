// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header1  should render 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#fff",
        "fontFamily": "Roboto",
        "fontSize": 36,
        "fontWeight": "bold",
        "padding": 20,
      },
      undefined,
    ]
  }
/>
`;
